File: CrypticPassword_1.5.0/android/tests/CrypticPasswordAlgorithmSmoke.java
import com.anoop.crypticpassword.CrypticPasswordAlgorithm;

public final class CrypticPasswordAlgorithmSmoke {
    public static void main(String[] args) {
        for (CrypticPasswordAlgorithm.Version version : CrypticPasswordAlgorithm.Version.values()) {
            String s = CrypticPasswordAlgorithm.shuffle(version, 2023);
            if (s.length() != CrypticPasswordAlgorithm.TOTAL_CELLS) {
                throw new IllegalStateException(version.label + " length=" + s.length());
            }
            System.out.println("PASS " + version.label + " " + s.substring(0, 12));
        }
        System.out.println("ANDROID_JAVA_CORE_PASS_ALL_ALGORITHMS");
    }
}

File: CrypticPassword_1.5.0/ios/tests/CrypticPasswordAlgorithmSmoke.swift
import Foundation

@main
struct CrypticPasswordAlgorithmSmoke {
    static func main() {
        let pins = [1, 2, 2023, 99999]
        for version in CrypticPasswordAlgorithm.Version.allCases {
            for pin in pins {
                let shuffled = CrypticPasswordAlgorithm.shuffle(version: version, pin: pin)
                if shuffled.count != CrypticPasswordAlgorithm.totalCells {
                    fatalError("Invalid shuffled length for \(version) pin \(pin): \(shuffled.count)")
                }
            }
        }
        print("IOS_SWIFT_CORE_PASS_ALL_ALGORITHMS")
    }
}
